File: 字符串/day2/最长公共前缀/longest_common_prefix_pfan8.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size() == 0) return "";
        if(strs.size() == 1) return strs[0];
        size_t min_len = INT_MAX;
        for(string s : strs) {
            min_len = min(min_len, s.size());
        }
        int word = 0;
        for(;(word < strs.size()) && (min_len > 0);) {
            if(strs[0].substr(0,min_len) == strs[word].substr(0,min_len)) {
                word++;
            } else {
                min_len--;
            }
        }
        return strs[0].substr(0,min_len);
    }
};

File: 字符串/day2/最长公共前缀/longest_common_prefix_pfan8.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        min_len = min([len(x) for x in strs])
        word = 0
        while word < len(strs) and min_len > 0:
            if strs[0][:min_len] == strs[word][:min_len]:
                word += 1
            else:
                min_len -= 1
        return strs[0][:min_len]